frb_codegen/src/library/codegen/generator/wire/rust/spec_generator/codec/cst/decoder/ty/enumeration.rs

```diff
@@ -93,21 +96,12 @@ impl<'a> WireRustCodecCstGeneratorDecoderTrait for EnumRefWireRustCodecCstGenera
     }
 
     fn generate_impl_new_with_nullptr(&self) -> Option<WireRustOutputCode> {
-        let src = self.ir.get(self.context.ir_pack);
-
-        let inflators = src
-            .variants()
-            .iter()
-            .filter_map(|variant| self.generate_impl_new_with_nullptr_variant(variant))
-            .collect_vec();
-
         Some(WireRustOutputCode {
             body: generate_impl_new_with_nullptr_code_block(
                 self.ir.clone(),
                 self.context,
-                "Self { tag: -1, kind: core::ptr::null_mut() }",
+                "Self { tag: -1, kind: core::mem::MaybeUninit::uninit() }",
```

Comment:
nit: currently, it seems that we use MaybeUninit + union. What if we do not use MaybeUninit and only use a union with one extra case. For example:

```
kind: MaybeUninit<SthKind>

pub union SthKind {
  apple: *mut Apple,
  orange: *mut Orange,
}
```

change to

```
kind: SthKind

pub union SthKind {
  nothing: (), // <--- And we init the union to this variant
  apple: *mut Apple,
  orange: *mut Orange,
}
```

Reply:
I think that would be similar to `MaybeUninit::zeroed()` so it would just be a stylistic choice, up to you. In fact this is how MaybeUninit is defined as well.

Reply:
> In fact this is how MaybeUninit is defined as well.

Totally agree - that's why I have this proposal ;)

> I think that would be similar to MaybeUninit::zeroed() so it would just be a stylistic choice, up to you. 

I personally want to avoid adding those Copy, thus removing MaybeUninit looks a bit helpful